test/python/circuit/test_gate_power.py

```diff
@@ -187,11 +207,45 @@ def test_invariant1_int(self, n):
     def test_invariant2(self, n):
         """Test op^(n) * op^(-n) == I"""
         result = Operator(SGate()).power(n) & Operator(SGate()).power(-n)
-        expected = Operator(eye(2))
+        expected = Operator(np.eye(2))
 
         self.assertEqual(len(result.data), len(expected.data))
         self.assertEqual(result, expected)
 
 
+@ddt
+class TestEfficientGatePowering(QiskitTestCase):
+    """Test gate powering is efficient where expected."""
+
+    @data(
+        (CPhaseGate(0.1), CPhaseGate),
+        (IGate(), IGate),
+        (PhaseGate(-0.1), PhaseGate),
+        (RGate(0.1, 0.1), RGate),
+        (RXGate(0.1), RXGate),
+        (RXXGate(-0.1), RXXGate),
+        (RYGate(-0.1), RYGate),
+        (RYYGate(0.1), RYYGate),
+        (RZGate(0.1), RZGate),
+        (RZXGate(-0.1), RZXGate),
+        (RZZGate(-0.1), RZZGate),
+        (SGate(), PhaseGate),
+        (TGate(), PhaseGate),
+        (XXMinusYYGate(-0.1, 0.1), XXMinusYYGate),
+        (XXPlusYYGate(0.1, 0.1), XXPlusYYGate),
+        (ZGate(), PhaseGate),
+        (iSwapGate(), XXPlusYYGate),
+    )
+    @unpack
+    def test_efficient_gate_powering(self, gate: Gate, output_gate_type: Type[Gate]):
+        """Test efficient gate powering."""
+        exponents = (-5, -0.5, -0.1, 0, 0.1, 0.5, 5)
+        for exponent in exponents:
+            result = gate.power(exponent)
+            assert isinstance(result, output_gate_type)
```

Comment:
Could you use the class method `self.assertIsInstance` instead of the assert statement, like in all the other places? 🙂